Clang-tidy: malloc 0 bytes in register.cpp

```
[133/246] Building CXX object CMakeFiles/DDNet-Server.dir/src/engine/server/register.cpp.o
FAILED: CMakeFiles/DDNet-Server.dir/src/engine/server/register.cpp.o 
/usr/bin/cmake -E __run_co_compile --tidy="clang-tidy;-warnings-as-errors=*;--extra-arg-before=--driver-mode=g++" --source=../src/engine/server/register.cpp -- /usr/bin/c++ -DCONF_DEBUG -DCONF_INFORM_UPDATE -DCONF_OPENSSL -DCONF_VIDEORECORDER -DGAME_RELEASE_VERSION=\"16.1\" -DGLEW_STATIC -Isrc -I../src -g -fdiagnostics-color=always -fstack-protector-all -fno-exceptions -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wformat=2 -Wno-nullability-completeness -Wno-tautological-constant-out-of-range-compare -Wduplicated-cond -Wduplicated-branches -Wlogical-op -Wrestrict -Wshadow=global -Wsuggest-override -std=gnu++17 -MD -MT CMakeFiles/DDNet-Server.dir/src/engine/server/register.cpp.o -MF CMakeFiles/DDNet-Server.dir/src/engine/server/register.cpp.o.d -o CMakeFiles/DDNet-Server.dir/src/engine/server/register.cpp.o -c ../src/engine/server/register.cpp
../src/engine/shared/http.h:108:30: error: Call to 'malloc' has an allocation size of 0 bytes [clang-analyzer-optin.portability.UnixAPI,-warnings-as-errors]
                m_pBody = (unsigned char *)malloc(DataLength);
                                           ^
/home/chiller/Desktop/git/ddnet/clang-tidy/../src/engine/server/register.cpp:564:5: note: Assuming the condition is false
        if((pPacket->m_Flags & NETSENDFLAG_CONNLESS) == 0)
           ^
/home/chiller/Desktop/git/ddnet/clang-tidy/../src/engine/server/register.cpp:564:2: note: Taking false branch
        if((pPacket->m_Flags & NETSENDFLAG_CONNLESS) == 0)
        ^
/home/chiller/Desktop/git/ddnet/clang-tidy/../src/engine/server/register.cpp:568:5: note: Assuming the condition is true
        if(pPacket->m_DataSize >= (int)sizeof(m_aVerifyPacketPrefix) &&
           ^
/home/chiller/Desktop/git/ddnet/clang-tidy/../src/engine/server/register.cpp:568:5: note: Left side of '&&' is true
/home/chiller/Desktop/git/ddnet/clang-tidy/../src/engine/server/register.cpp:569:3: note: Assuming the condition is true
                mem_comp(pPacket->m_pData, m_aVerifyPacketPrefix, sizeof(m_aVerifyPacketPrefix)) == 0)
                ^
/home/chiller/Desktop/git/ddnet/clang-tidy/../src/engine/server/register.cpp:568:2: note: Taking true branch
        if(pPacket->m_DataSize >= (int)sizeof(m_aVerifyPacketPrefix) &&
        ^
/home/chiller/Desktop/git/ddnet/clang-tidy/../src/engine/server/register.cpp:576:6: note: Assuming the condition is false
                if(Unpacker.Error())
                   ^
/home/chiller/Desktop/git/ddnet/clang-tidy/../src/engine/server/register.cpp:576:3: note: Taking false branch
                if(Unpacker.Error())
                ^
/home/chiller/Desktop/git/ddnet/clang-tidy/../src/engine/server/register.cpp:584:3: note: Taking false branch
                if(ProtocolFromString(&Protocol, pProtocol))
                ^
/home/chiller/Desktop/git/ddnet/clang-tidy/../src/engine/server/register.cpp:589:3: note: Calling 'CProtocol::OnToken'
                m_aProtocols[Protocol].OnToken(pToken);
                ^
/home/chiller/Desktop/git/ddnet/clang-tidy/../src/engine/server/register.cpp:362:5: note: Assuming the condition is true
        if(time_get() >= m_NextRegister)
           ^
/home/chiller/Desktop/git/ddnet/clang-tidy/../src/engine/server/register.cpp:362:2: note: Taking true branch
        if(time_get() >= m_NextRegister)
        ^
/home/chiller/Desktop/git/ddnet/clang-tidy/../src/engine/server/register.cpp:364:3: note: Calling 'CProtocol::SendRegister'
                SendRegister();
                ^
/home/chiller/Desktop/git/ddnet/clang-tidy/../src/engine/server/register.cpp:269:18: note: Assuming 'InfoSerial' is <= field 'm_LatestSuccessfulInfoSerial'
        bool SendInfo = InfoSerial > m_pShared->m_pGlobal->m_LatestSuccessfulInfoSerial;
                        ^
/home/chiller/Desktop/git/ddnet/clang-tidy/../src/engine/server/register.cpp:273:5: note: 'SendInfo' is false
        if(SendInfo)
           ^
/home/chiller/Desktop/git/ddnet/clang-tidy/../src/engine/server/register.cpp:273:2: note: Taking false branch
        if(SendInfo)
        ^
/home/chiller/Desktop/git/ddnet/clang-tidy/../src/engine/server/register.cpp:279:84: note: Passing the value 0 via 3rd parameter 'DataLength'
                pRegister = HttpPost(m_pParent->m_pConfig->m_SvRegisterUrl, (unsigned char *)"", 0);
                                                                                                 ^
/home/chiller/Desktop/git/ddnet/clang-tidy/../src/engine/server/register.cpp:279:15: note: Calling 'HttpPost'
                pRegister = HttpPost(m_pParent->m_pConfig->m_SvRegisterUrl, (unsigned char *)"", 0);
                            ^
../src/engine/shared/http.h:177:23: note: Passing the value 0 via 2nd parameter 'DataLength'
        pResult->Post(pData, DataLength);
                             ^
../src/engine/shared/http.h:177:2: note: Calling 'CHttpRequest::Post'
        pResult->Post(pData, DataLength);
        ^
../src/engine/shared/http.h:108:30: note: Call to 'malloc' has an allocation size of 0 bytes
                m_pBody = (unsigned char *)malloc(DataLength);
                                           ^
415 warnings generated.
Suppressed 414 warnings (409 in non-user code, 5 with check filters).
Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
1 warning treated as error
```

--- a/src/engine/shared/http.h
+++ b/src/engine/shared/http.h
@@ -1,6 +1,7 @@
 #ifndef ENGINE_SHARED_HTTP_H
 #define ENGINE_SHARED_HTTP_H
 
+#include <algorithm>
 #include <atomic>
 #include <engine/shared/jobs.h>
 
@@ -105,7 +106,7 @@ public:
 	{
 		m_Type = REQUEST::POST;
 		m_BodyLength = DataLength;
-		m_pBody = (unsigned char *)malloc(DataLength);
+		m_pBody = (unsigned char *)malloc(std::max((size_t)1, DataLength));
 		mem_copy(m_pBody, pData, DataLength);
 	}
 	void PostJson(const char *pJson)

